lib/presentation/molecules/tip_resource_molecule.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/material.dart';
+import 'package:infinite_horizons/presentation/atoms/separator_atom.dart';
+import 'package:infinite_horizons/presentation/core/global_variables.dart';
+import 'package:infinite_horizons/presentation/molecules/molecules.dart';
+
+class TipResourceMolecule extends StatefulWidget {
+  const TipResourceMolecule({
+    this.innerWidget,
+    required this.title,
+    super.key,
+    this.topBarType = TopBarType.back,
+    this.separatorVariant = SeparatorVariant.closeWidgets,
+    this.isYouTube = false,
+    this.url,
+  });
+  final Widget? innerWidget;
+  final TopBarType topBarType;
+  final String title;
+  final SeparatorVariant separatorVariant;
+  final bool isYouTube;
+  final String? url;
+  @override
+  State<TipResourceMolecule> createState() => _TipResourceMoleculeState();
+}
+
+class _TipResourceMoleculeState extends State<TipResourceMolecule> {
+  bool isFullScreen = false;
+  EdgeInsetsGeometry get padding {
+    if (isFullScreen) {
+      return EdgeInsets.zero;
+    }
+    return GlobalVariables.defaultPadding;
+  }
+
+  void listener() {
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Padding(
+        padding: padding,
+        child: Column(
+          children: [
+            if (!isFullScreen)
+              TopBarMolecule(
+                topBarType: widget.topBarType,
+                title: widget.title,
+              ),
+            SeparatorAtom(variant: widget.separatorVariant),
+            Expanded(
+                child: widget.isYouTube
+                    ? YoutubePlayerMolecule(
+                        url: widget.url!,
+                        callback: (bool ytFullScreen) {
+                          setState(() {
+                            isFullScreen = ytFullScreen;
+                          });
+                        },
+                      )
```

Comment:
Why youtube player isen't open on a new page?, it should not be part of the tip resource molecule.
YouTube is an entirely different page and not a popup inside of the tip resource molecule.

Reply:
If you don't want to use a Global Variable and still want to keep the YouTubePlayer Molecule as just the player then this is the only other way because the Top Bar we need to disappear on entering full screen is part of the TipResourceMolecule and we are calling that in the TipResourcesPage and then seinding a nested call to YoutubePlayerMolecule

Reply:
A total different page called youtube_page.dart, when clicking on a YouTube link the user gets navigated to a different page.
And YouTubePage will manage the page topbar if needed and contain YoutubePlayerMolecule